### ISSUE TITLE
console history is not cleared between requests

### ISSUE THREAD

**Issue by zetachang** [reactjs/react-rails#625]
My app doesn't crash, but I'm getting unexpected behavior. So, I've described the unexpected behavior and suggested a new behavior.

---

`console.history` is not cleared when used with prerender method using `therubyracer` runtime. 

Here is an example extracted from the [sprockets_renderer_test](https://github.com/reactjs/react-rails/blob/d2cc6045ab460bbb667c51fd500c49dd37de21c5/test/react/server_rendering/sprockets_renderer_test.rb#L26)

``` rb
@renderer.render("TodoListWithConsoleLog", {todos: ["log some messages"]}, nil)
puts @renderer.render("TodoListWithConsoleLog", {todos: ["log some messages"]}, nil)
```

will result to

``` html
<ul data-reactroot="" data-reactid="1" data-react-checksum="-1309656939"><li data-reactid="2">Console Logged</li><li id="status" data-reactid="3">yep</li><li data-reactid="4">log some messages</li></ul>
<script>
console.log.apply(console, ["got initial state"]);
console.warn.apply(console, ["mounted component"]);
console.error.apply(console, ["rendered!","foo"]);
console.log.apply(console, ["got initial state"]);
console.warn.apply(console, ["mounted component"]);
console.error.apply(console, ["rendered!","foo"]);
</script>
```

The rubyracer `v8_context` deep inside is the same during multiple `eval`s, and so is other stuff user might stored in `global`.

This is not happened in the default execjs runtime given that [it shell out every time `eval` invoked.](https://github.com/rails/execjs/blob/master/lib/execjs/external_runtime.rb#L171)

Test environment
- `react-rails`: master
- `therubyracer`: 0.12.2

**Comment by rmosolgo** [comment]
In general, I think reusing a a context is OK, otherwise, you pay overhead for starting up a new one every time. 

Maybe it's time to dust off https://github.com/reactjs/react-rails/pull/559

**Comment by zetachang** [comment]
Yeah, agreed on the overhead to start up new one every time, but at least we could clear up `console`, `global`, `window`.. ?

**Comment by rmosolgo** [comment]
Addressed in #692, will be fixed in 2.0.0
